httpcore/_backends/base.py

```diff
@@ -1,13 +1,19 @@
 import ssl
 import time
 import typing
+from types import TracebackType
 
 SOCKET_OPTION = typing.Union[
     typing.Tuple[int, int, int],
     typing.Tuple[int, int, typing.Union[bytes, bytearray]],
     typing.Tuple[int, int, None, int],
 ]
 
+_SELF_NETWORK_STREAM = typing.TypeVar("_SELF_NETWORK_STREAM", bound="NetworkStream")
+_SELF_ASYNC_NETWORK_STREAM = typing.TypeVar(
+    "_SELF_ASYNC_NETWORK_STREAM", bound="AsyncNetworkStream"
+)
```

Comment:
Perhaps we should add `typing_extensions` as a CI requirement so we can just use `Self`? We'd have to import under a `if TYPE_CHECKING` block but that seems okay.

Reply:
I think `typing_extensions` is fine to have as a dependency. I would like to say say it's widely used.